meta/data.go

```diff
@@ -74,14 +74,71 @@ func (data *Data) CreateNode(host string) error {
 }
 
 // DeleteNode removes a node from the metadata.
-func (data *Data) DeleteNode(id uint64) error {
-	for i := range data.Nodes {
-		if data.Nodes[i].ID == id {
-			data.Nodes = append(data.Nodes[:i], data.Nodes[i+1:]...)
-			return nil
+func (data *Data) DeleteNode(id uint64, force bool) error {
+	// Node has to be larger than 0 to be real
+	if id == 0 {
+		return ErrNodeIDRequired
+	}
+	// Is this a valid node?
+	nodeInfo := data.Node(id)
+	if nodeInfo == nil {
+		return ErrNodeNotFound
+	}
+	// Am I the only node?  If so, nothing to do
+	if len(data.Nodes) == 1 {
+		return ErrNodeUnableToDropFinalNode
+	}
+
+	// Determine if there are any any non-replicated nodes and force was not specified
+	if !force {
+		for _, di := range data.Databases {
+			for _, rp := range di.RetentionPolicies {
+				// ignore replicated retention policies
```

Comment:
Purely a style concern, but if you instead do:
```
if rp.ReplicaN > 1 {
    continue
}
```
you avoid a level of indentation and, I think, make the code a little easier to read.